lib/acme/resources/challenges/tlssni01.rb

```diff
@@ -0,0 +1,24 @@
+class Acme::Resources::Challenges::TLSSNI01 < Acme::Resources::Challenges::Base
+
+  def hostname
+    hd = crypto.digest.hexdigest(authorization_key)
+    "#{hd[0..31]}.#{hd[32..64]}.acme.invalid"
+  end
+
+  def temporary_cert
+    @temporary_cert ||= client.new_self_signed_certificate self.hostname 
+  end
+
+  def self_signed_certificate
+    self.temporary_cert.to_pem
+  end
```

Comment:
Probably not. However, the `create_self_signed_cert` itself supports multiple hostnames: The first one is used in `CN=` and all hostnames are used as SAN names.